packages/web/pages/stake/index.tsx

```diff
@@ -71,33 +85,94 @@ export const Staking: React.FC = observer(() => {
     return { currency: osmo, amount: primitiveAmount.amount, denom: osmo };
   }, [osmo, primitiveAmount]);
 
+  const delegationQuery = cosmosQueries.queryDelegations.getQueryBech32Address(
+    account?.address ?? ""
+  );
+
+  const userValidatorDelegations = delegationQuery.delegations;
+
+  const usersValidatorsMap = useMemo(() => {
+    const delegationsMap = new Map<string, StakingType.Delegation>();
+
+    userValidatorDelegations.forEach((delegation: StakingType.Delegation) => {
+      delegationsMap.set(delegation.delegation.validator_address, delegation);
+    });
+
+    return delegationsMap;
+  }, [userValidatorDelegations]);
+
+  const [amountDefault, setAmountDefault] = useState<AmountDefault>("input");
+
+  const getProperties = useCallback(() => {
+    const amountUSD = priceStore
+      .calculatePrice(
+        new CoinPretty(
+          osmo,
+          new Dec(amountConfig.amount).mul(
+            DecUtils.getTenExponentNInPrecisionRange(osmo.coinDecimals)
+          )
```

Comment:
I believe CoinPretty accepts the amount without decimals. So we could do `amountConfig.toCoinPrimitive().amount` as the second arg.

Reply:
Please test this and confirm it works.

Reply:
Also, similar to the other query data, we shouldn't need a callback here. We can just get the USD value of the current amount in the top level. The price store is observable, so if price data is refreshed, this component would rerender and the USD amount would update.